core/templates/pages/story-editor-page/services/story-editor-navigation.service.ts

```diff
@@ -30,28 +30,35 @@ const STORY_PREVIEW = 'story_preview';
 })
 export class StoryEditorNavigationService {
     activeTab: string = 'story_editor';
-    chapterId: string = null;
-    chapterIndex: number = null;
+    chapterId!: string;
+    // 'chapterIndex' is null when we are navigating to a chapter with its ID.
+    chapterIndex!: number | null;
+
     constructor(private windowRef: WindowRef) {}
 
     getActiveTab(): string {
       return this.activeTab;
     }
+
     setChapterId(id: string): void {
       this.chapterId = id;
     }
-    getChapterIndex(): number {
+
+    getChapterIndex(): number | null {
       return this.chapterIndex;
     }
+
     getChapterId(): string {
       return this.chapterId;
     }
-    navigateToChapterEditorWithId(id: string, index: number): void {
+
+    navigateToChapterEditorWithId(id: string, index: number | null): void {
```

Comment:
```suggestion
    navigateToChapterEditorWithId(id: string, index!: number): void {
```

Reply:
These are arguments, we don't want to do non-null assertion here since we're not doing initialization here, we get these properties at the time of function calling.